src/libcore/array.rs

```diff
@@ -264,6 +267,269 @@ macro_rules! array_impls {
     }
 }
 
+#[cfg(not(bootstrap))]
+mod impls_using_const_generics {
+    use super::*;
+
+    #[stable(feature = "rust1", since = "1.0.0")]
+    impl<T, const N: usize> AsRef<[T]> for [T; N]
+        where [(); N]: AllowedArrayLength
```

Comment:
Also what's with the `()` here? (Leave a comment in the code about it with the rationale)

Reply:
I had the same thought, but I’d suggest trying to make it `T` to make it more obvious for readers of the docs.

Reply:
Switched this to use the "real" type instead of `()`.